uamqp/client.py

```diff
@@ -373,12 +373,25 @@ def _transfer_message(self, message_delivery, timeout=0):
     def _on_send_complete(self, message_delivery, message, reason, state):
         # TODO: check whether the callback would be called in case of message expiry or link going down
         # and if so handle the state in the callback
-        if SEND_DISPOSITION_ACCEPT in state:
+        if state and SEND_DISPOSITION_ACCEPT in state:
             message_delivery.state = MessageDeliveryState.Ok
         else:
             # TODO: sending disposition state could only be rejected/accepted?
-            message_delivery.state = MessageDeliveryState.Error
-            message_delivery.reason = reason
+            # TODO: error action
+            if not state and reason == LinkDeliverySettleReason.NotDelivered:
+                # TODO: message is not delivered
+                message_delivery.state = MessageDeliveryState.Error
+                return
+            error_response = ErrorResponse(state[SEND_DISPOSITION_REJECT])
+            if error_response.condition == b'com.microsoft:server-busy':
+                # TODO: max retries configs
+                time.sleep(4)  # 4 is what we're doing nowadays in EH/SB, service tells client to backoff for 4 seconds
```

Comment:
We should look at ways of pulling this logic out (serverbusy + backoff) - as I believe it's EH/SB specific and I really don't like how I added it into the uamqp-c version. We can keep it here while we conduct the perf tests, but should look at better designs for this before adding it.

Reply:
it's completely on me being too lazy -- I just hacked the code to get it work for perf analysis first.

in uamqp this logic is already pulled out into ErrorPolicy/ErrorAction. I have updated the implementation to reuse our code in uamqp.

I do want la better design for the error handling/retry part, but I think the current uamqp way is be a good start.

Reply:
I'd actually rather we just open an issue to find a design for it than resue the old ErrorPolicy/ErrorAction code.... at least not in its current form. So just hardcoding the error with the sleep with a big bright TODO next to it might be fine.

Reply:
issue logged: https://github.com/Azure/azure-sdk-for-python/issues/20654